Add and edits readmes (#47)
loambuild/loam-sdk: crates/loam-sdk/README.md

# loam-sdk

- [Getting Started](#getting-started)
    - [Installation](#installation)
    - [Setup](#setup)
    - [Redeploy](#redeploy)
- [Subcontracts](#subcontracts)
    - [Creating Contract Subcontracts](#creating-contract-subcontracts)
    - [External API](#external-api)

> - [CoreSubcontract](#coreriff)

Reviewer: ```suggestion
- [Core](#core-subcontract)
```

> -   [Using the Core Subcontract](#using-the-coreriff)

Reviewer: ```suggestion
-   [Using the Core Subcontract](#using-the-core-subcontract)
```

### Installation

To install `just`, run the following command:

```bash
cargo install just
```

### Setup

To set up the environment, run:

```bash
just setup
```

### Redeploy

To see redeployment in action, use:

```bash
just redeploy

> ```

Reviewer 2: @willemneal this seems like stuff that would be better in a `CONTRIBUTING.md` file. That is, it seems like info about how to contribute to `loam-sdk`, not how to install it. Is that correct?

## Subcontracts

A subcontract is a type that implements the `IntoKey` trait, which is used for lazily loading and storing the type.

### Creating  Subcontracts

> Here's an example of how to create a contract riff:

Reviewer 2: The word "riff" is used throughout this file. Let's update it all to "subcontract".

> Here's an example of how to create a contract riff:

Reviewer: ```suggestion
Here's an example of how to create a subcontract:
```

```rust
#[contracttype]
#[derive(IntoKey)]
pub struct Messages(Map<Address, String>);
```

This generates the following implementation:

```rust
impl IntoKey for Messages {
    type Key = IntoVal<Env, RawVal>;
    fn into_key() -> Self::Key {
      String::from_slice("messages")
    }
```

### External API

You can also create and implement external APIs for contract subcontracts:

```rust
#[riff]
pub trait IsPostable {
    fn messages_get(&self, author: Address) -> Option<String>;
    fn messages_set(&mut self, author: Address, text: String);
}
```

## Core Subcontract

The `Core` trait provides the minimum logic needed for a contract to be redeployable. A contract should be able to be redeployed to another contract that can also be redeployed. Redeployment requires admin status, as it would be undesirable for an account to redeploy the contract without permission.

> ### Using the CoreSubcontract

Reviewer: ```suggestion
### Using  `Core`
```

> To use the core riff, create a `Contract` structure and implement the `Core` for it. The `Contract` will be redeployable and will be able to implement other Subcontracts.

Reviewer: ```suggestion
To use the core subcontract, create a `Contract` struct and implement the `Core` for it. This makes `Contract` redeployable by the Admin of the contract and will continue to be redeployable if the new contract also implements `Core`. After `Core` other Subcontracts can be added as needed.
```

```rust
use loam_sdk::{soroban_contract, soroban_sdk};

> use loam_sdk_core_riff::{owner::Owner, CoreSubcontract};

Reviewer: ```suggestion
use loam_subcontract_core::{Admin, Core};
```

pub struct Contract;

> impl CoreSubcontract for Contract {

Reviewer: ```suggestion
impl Core for Contract {
```
